File: algorithms/easy/left_and_right_sum_differences.ex
defmodule Solution do
  @spec left_right_difference(nums :: [integer]) :: [integer]
  def left_right_difference(nums) do
    right = Enum.sum(nums)

    nums
    |> Enum.reduce({[], 0, right}, fn n, {acc, left, right} ->
      {[abs(right - left - n) | acc], left + n, right - n}
    end)
    |> elem(0)
    |> Enum.reverse()
  end
end

File: algorithms/easy/left_and_right_sum_differences.rs
impl Solution {
    pub fn left_right_difference(nums: Vec<i32>) -> Vec<i32> {
        let mut right = nums.iter().sum::<i32>();
        let mut left = 0;
        let mut res = vec![0; nums.len()];
        for (i, n) in nums.into_iter().enumerate() {
            res[i] = (right - left - n).abs();
            left += n;
            right -= n;
        }
        res
    }
}
